src/context/video-context.js

```diff
@@ -0,0 +1,60 @@
+import { createContext, useContext, useEffect, useReducer } from "react";
+import axios from "axios";
+const initialState = {
+  videos: [],
+  categories: [],
+  selectedCategory: "All",
+};
+
+const VideoContext = createContext();
+const VideoProvider = ({ children }) => {
+  const videoReducer = (videoState, { type, payload }) => {
```

Comment:
You can make folder reducer and put videoReducer in videoreducer file.

Reply:
okay